packages/extension-vscode/src/quickFixProvider.ts

```diff
@@ -0,0 +1,77 @@
+import { CodeAction, CodeActionKind, CodeActionParams, Command, TextDocuments } from 'vscode-languageserver';
+import { TextDocument } from 'vscode-languageserver-textdocument';
+import { getProblemNameFromDiagnostic } from './utils/problems';
+
+export class QuickFixActionProvider {
+    private documents: TextDocuments<TextDocument>;
+    private sourceName: string;
+
+    private quickFixActionsTemplates = new Map([
+        ['vscode-webhint/ignore-hint-project', (category: string) => {
+            return `Ignore hint '${category}' in this project`;
+        }],
+        ['vscode-webhint/ignore-hint-global', (category: string) => {
+            return `Ignore hint '${category}' in all projects`;
+        }],
+        ['vscode-webhint/ignore-problem-project', (name: string) => {
+            return `Ignore problem '${name}' in this project`;
```

Comment:
These will need to be scoped to only the `compat-api` hints (`compat-api/css` and `compat-api/html`). Currently it matches other hint messages so things like `css-prefix-order` end up with an ignore entry, but the generated `.hintrc` doesn't work (as that hint doesn't accept an ignore list).